src/test/fuzz/banman.cpp

```diff
@@ -93,6 +95,22 @@ FUZZ_TARGET_INIT(banman, initialize_banman)
                     ban_man.Discourage(ConsumeNetAddr(fuzzed_data_provider));
                 });
         }
+        if (!force_read_and_write_to_err) {
+            ban_man.DumpBanlist();
+            SetMockTime(ConsumeTime(fuzzed_data_provider));
+            banmap_t banmap;
+            ban_man.GetBanned(banmap);
+            BanMan ban_man_read{banlist_file, nullptr, offset};
+            banmap_t banmap_read;
+            ban_man_read.GetBanned(banmap_read);
```

Comment:
The dumped data may end up containing more entries than `banmap` and `banmap_read` because `SweepBanned()` may leave them when called from `DumpBanlist()` but when called from `GetBanned()` after `SetMockTime()` it may remove "expired" entries.

It is ok since both `banmap` and `banmap_read` will contain the same, but if somebody ends up debugging this and looking at the file, then it may be confusing. Maybe swap lines 99 and 100.

Reply:
This was done on purpose to check `SweepBanned` is called on read

Reply:
Hmm, then maybe add one entry that will certainly expire? This will also make the intention more explicit:

```diff
diff --git i/src/test/fuzz/banman.cpp w/src/test/fuzz/banman.cpp
index 6aa95b484f..b3b10da084 100644
--- i/src/test/fuzz/banman.cpp
+++ w/src/test/fuzz/banman.cpp
@@ -93,16 +93,23 @@ FUZZ_TARGET_INIT(banman, initialize_banman)
                 },
                 [&] {
                     ban_man.Discourage(ConsumeNetAddr(fuzzed_data_provider));
                 });
         }
         if (!force_read_and_write_to_err) {
+            const auto mock_time = ConsumeTime(fuzzed_data_provider);
+            SetMockTime(mock_time);
+            // Add one ban entry that will expire to check that on read expired entries are removed.
+            ban_man.Ban(ConsumeNetAddr(fuzzed_data_provider), mock_time + 5, true);
             ban_man.DumpBanlist();
-            SetMockTime(ConsumeTime(fuzzed_data_provider));
+            banmap_t banmap_unexpired;
+            ban_man.GetBanned(banmap_unexpired);
+            SetMockTime(mock_time + 10);
             banmap_t banmap;
             ban_man.GetBanned(banmap);
+            assert(banmap.size() < banmap_unexpired.size()); // At least one should have expired.
             BanMan ban_man_read{banlist_file, nullptr, offset};
             banmap_t banmap_read;
             ban_man_read.GetBanned(banmap_read);
             assert(banmap.size() == banmap_read.size());
             auto it1{banmap.begin()};
             auto it2{banmap_read.begin()};
```